src/authentication_middleware.js

```diff
@@ -0,0 +1,9 @@
+import constants from './constants';
+
+export default (socket) => (store) => (next) => (action) => {
+  if (action.type === constants.LOGIN) {
+    socket.emit('authentication', { pin: action.pin });
```

Comment:
Can probably just return out of the middleware since it won't need to go to the next middleware or reducer.